juniper_official/Protocols/Mpls/check-lsp-flap-count-oc.rule

```diff
@@ -0,0 +1,104 @@
+/*
+ * Detects MPLS LSP state changes and notifies when state change count 
+ * increases frequently.
+ * One inputs control detection
+ * 
+ *   1) input-lsp-name, is a regular expression that matches the LSP name that
+ *      you would like to monitor.  By default it's '.*', which matches all
+ *      LSPs. Use something like 'LSP1|LSP2' to match only LSP1 and LSP2.
+ */
+ healthbot {
+    topic protocol.mpls {
+        synopsis "MPLS statistics analyzer";
+        description "Monitors MPLS LSP and RSVP TE statistics and notifies anomalies";
+        rule check-lsp-flap-count-oc {
+            /*
+             * Use LSP name as key for rule.
+             */
+            keys lsp-name;
+            synopsis "LSP state analyzer";
+            description "Collects LSP stats periodically and notifies when LSP flaps";
+            /*
+             * Sensor configuration to get data from network devices.
+             */
+            sensor lsp {
+                open-config {
+                    sensor-name /network-instances/network-instance/mpls/lsps;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined using sensor path. Map the longer sensor names
+             * to the shorter field names used in the rules.
+             */
+            field lsp-name {
+                sensor lsp {
+                    where "/network-instances/network-instance/mpls/lsps/constrained-path/tunnels/tunnel/@name =~ /{{input-lsp-name}}/";
+                    path "/network-instances/network-instance/mpls/lsps/constrained-path/tunnels/tunnel/@name";
+                }
+                type string;
+                description "LSP name to be monitored";
+            }
+            field lsp-state-change-count {
+                sensor lsp {
+                    path /network-instances/network-instance/mpls/lsps/constrained-path/tunnels/tunnel/state/counters/state-changes;
+                }
+                description "LSP state change count";
+            }
+            /*
+             * Anomaly detection logic.
+             */
+            trigger lsp-flap-count {
+                frequency 90s;
+                term lsp-stable {
+                    when {
+                        exists "$lsp-name";
+                        increasing-at-most-by-value "$lsp-state-change-count" {
+                            value 1;
```

Comment:
Should this be 0? As in we don't want the count to increase at all, right?

Reply:
Yes, One makes sense. changed